bioio/bio_image.py

```diff
@@ -1174,3 +1227,63 @@ def imread(
         np.ndarray.
     """
     return _construct_img(image, scene_id, **kwargs).data
+
+
+from typing import Any, Dict
+
+
+def _check_plugin_support(
+    plugin: PluginEntry, image: biob.types.ImageLike, fs_kwargs: Dict[str, Any] = {}
+) -> Dict[str, Union[bool, Any]]:
+    """Helper function to check if a plugin supports the given image."""
+    try:
+        ReaderClass = plugin.metadata.get_reader()
+        return {
+            "supported": ReaderClass.is_supported_image(
+                image=image, fs_kwargs=fs_kwargs
+            ),
+            "error": None,
+        }
+    except Exception as e:
+        return {"supported": False, "error": str(e)}
```

Comment:
The "supported", "error" dict is used enough that it should be a class.
```suggestion
from dataclasses import dataclass

@dataclass
class PluginSupport:
    supported: bool
    error: Optional[str]

def _check_plugin_support(
    plugin: PluginEntry, image: biob.types.ImageLike, fs_kwargs: Dict[str, Any] = {}
) -> Dict[str, PluginSupport]:
    """Helper function to check if a plugin supports the given image."""
    try:
        ReaderClass = plugin.metadata.get_reader()
        return PluginSupport(
            supported=ReaderClass.is_supported_image(
                image=image, fs_kwargs=fs_kwargs
            ),
            error=None,
        )
    except Exception as e:
        return PluginSupport(supported=False, error=str(e))
```

(This code snippet is untested, do not assume it works)